psh/psh.c

```diff
@@ -77,7 +77,7 @@ const psh_appentry_t *psh_findapp(char *appname)
 }
 
 
-static char *psh_stralloc(char *oldstr, const char *str)
+char *psh_stralloc(char *oldstr, const char *str)
```

Comment:
A slightly simpler version of `psh_stralloc()`
```c
char *psh_stralloc_t(char *oldstr, const char *str)
{
	char *newstr = strdup(str);
	
	if (newstr != NULL) {
		free(oldstr);
	}

	return newstr;
}
```

Reply:
I like it.